### PR TITLE
Customer Issue for token endpoint resolved and added the get schema b…

### PR DESCRIPTION
**Pull request by Shubhangi-cs**
…utton in the UI to generate the schema.

### CODE REVIEW

File: widgets/Ariba-batchsource.json [hunk 1]
```diff
@@ -31,6 +31,22 @@
             "connectionType": "Ariba"
           }
         },
+        {
+          "widget-type": "textbox",
+          "label": "Token Endpoint",
+          "name": "tokenURL",
+          "widget-attributes": {
+            "placeholder": "Token path of the Ariba API, for example, https://api.au.cloud.ariba.com"
```
Comment (dhiraj0116) [inline widgets/Ariba-batchsource.json]: Change token to access token

File: widgets/Ariba-batchsource.json [hunk 2]
```diff
@@ -31,6 +31,22 @@
...
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "API Endpoint",
+          "name": "baseURL",
+          "widget-attributes": {
+            "placeholder": "Base path of the Ariba API, for example, https://sandbox.api.sap.com"
```
Comment (dhiraj0116) [inline widgets/Ariba-batchsource.json]: base path -> base url

File: widgets/Ariba-connector.json [hunk 3]
```diff
@@ -5,8 +5,24 @@
...
     {
-      "label": "Credentials-updated",
+      "label": "Credentials",
       "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Token Endpoint",
+          "name": "tokenURL",
+          "widget-attributes": {
+            "placeholder": "Token path of the Ariba API, for example, https://api.au.cloud.ariba.com"
```
Comment (dhiraj0116) [inline widgets/Ariba-connector.json]: same as above comment

File: widgets/Ariba-connector.json [hunk 4]
```diff
@@ -5,8 +5,24 @@
...
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "API Endpoint",
+          "name": "baseURL",
+          "widget-attributes": {
+            "placeholder": "Base path of the Ariba API, for example, https://sandbox.api.sap.com"
```
Comment (dhiraj0116) [inline widgets/Ariba-connector.json]: same as above comment

File: src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java [hunk 5]
```diff
@@ -154,11 +167,25 @@ public final void validateToken(FailureCollector collector) {
...
     }
   }
 
+  private void getErrorFromResponse(AribaServices aribaServices, Response response, FailureCollector collector)
+    throws IOException {
+    ObjectMapper objectMapper = new ObjectMapper();
+    AribaResponseContainer responseContainer = aribaServices.tokenResponse(response);
+    if (responseContainer.getResponseBody() != null) {
+      InputStream responseStream = responseContainer.getResponseBody();
+      String errResponse = objectMapper.readTree(responseStream).get(MESSAGE).asText();
```
Comment (dhiraj0116) [inline src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java:185]: Double check that 'asText()' should not throw null pointer exception.

File: docs/Ariba-batchsource.md [hunk 6]
```diff
@@ -18,6 +18,7 @@ You can configure the following properties for the Ariba.
 
 **Reference Name:** Name used to uniquely identify this source for lineage,
 annotating metadata, etc.  
+**Token Endpoint (M)**:Token path of the Ariba API.
```
Comment (anup-codex) [inline docs/Ariba-batchsource.md]: Please be consistent with the formatting.

File: docs/Ariba-connector.md [hunk 7]
```diff
@@ -11,14 +11,16 @@ Properties
 
 **Description:** Description of the connection.
 
+**Token Endpoint (M)**:Token path of the Ariba API.
```
Comment (anup-codex) [inline docs/Ariba-connector.md]: +1

File: src/test/java/io/cdap/plugin/ariba/source/AribaBatchSourceTest.java [hunk 8]
```diff
@@ -189,7 +192,7 @@ public void testValidateAdvancedParametersError() {
       List<ValidationFailure> failures = s.getFailures();
       Assert.assertEquals("Failures size does not match, " +
                             "'CDF_ARIBA_01501 - Failed to call given Ariba service.'",
-                          1, failures.size());
+                          3, failures.size());
```
Comment (anup-codex) [inline src/test/java/io/cdap/plugin/ariba/source/AribaBatchSourceTest.java:195]: why asserting size to `3` and not `1`?

**Review by anup-codex — COMMENTED** [review]
Please check the `code formatting` and let me know what is the current code coverage %age.

File: widgets/Ariba-batchsource.json (hunk 1 above)
Comment (Shubhangi-cs) [reply widgets/Ariba-batchsource.json]: done

File: widgets/Ariba-batchsource.json (hunk 2 above)
Comment (Shubhangi-cs) [reply widgets/Ariba-batchsource.json]: done

File: widgets/Ariba-connector.json (hunk 3 above)
Comment (Shubhangi-cs) [reply widgets/Ariba-connector.json]: done

File: widgets/Ariba-connector.json (hunk 4 above)
Comment (Shubhangi-cs) [reply widgets/Ariba-connector.json]: done

File: docs/Ariba-batchsource.md (hunk 6 above)
Comment (Shubhangi-cs) [reply docs/Ariba-batchsource.md]: done the changes

File: docs/Ariba-connector.md (hunk 7 above)
Comment (Shubhangi-cs) [reply docs/Ariba-connector.md]: done the changes

File: src/test/java/io/cdap/plugin/ariba/source/AribaBatchSourceTest.java (hunk 8 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/ariba/source/AribaBatchSourceTest.java:195]: Acc to the previous code if credentials are incorrect it was throwing single error for all the incorrect credentials fields. But now the failures are being segregated, so it is throwing 3 failures for clientId, clientSecret and apiKey.

**Comment by Shubhangi-cs** [comment]
> Please check the `code formatting` and let me know what is the current code coverage %age.

the current code coverage is 79%.

File: docs/Ariba-batchsource.md [hunk 9]
```diff
@@ -10,32 +10,32 @@ For more information, see [Connect to SAP Ariba](https://cloud.google.com/data-f
...
 ## Basic 
 
 **Reference Name:** Name used to uniquely identify this source for lineage,
 annotating metadata, etc.  
-**Token Endpoint (M)**:Token path of the Ariba API.
-**API Endpoint (M)**: Base path of the Ariba API.    
-**System Type**: Type of system the Ariba instance is running on: Production or Sandbox.  
-**Realm (M)**: Realm name from which the data is to be extracted, for example, S4VALL.   
-**View Template Name (M)**: Name of the view template from which data is to be extracted, for example, RequestFactSystemView.
+**Token Endpoint (M):**Token path of the Ariba API.
```
Comment (anup-codex) [inline docs/Ariba-batchsource.md]: `**Token` -> `** Token` (formatting)

File: docs/Ariba-connector.md [hunk 10]
```diff
@@ -11,19 +11,19 @@ Properties
 
 **Description:** Description of the connection.
 
-**Token Endpoint (M)**:Token path of the Ariba API.
+**Token Endpoint (M):**Token path of the Ariba API.
```
Comment (anup-codex) [inline docs/Ariba-connector.md]: +1

**Review by anup-codex — COMMENTED** [review]
looks good, just few formatting left.

File: docs/Ariba-batchsource.md (hunk 9 above)
Comment (Shubhangi-cs) [reply docs/Ariba-batchsource.md]: Done

File: docs/Ariba-connector.md (hunk 10 above)
Comment (Shubhangi-cs) [reply docs/Ariba-connector.md]: Done

**Pull request merged**